rcp/org.testeditor.rcp4.views.teststepselection/src/org/testeditor/rcp4/views/teststepselector/TestStepSelector.xtend

```diff
@@ -27,35 +28,51 @@ import org.eclipse.swt.dnd.DragSourceListener
 import org.eclipse.swt.dnd.TextTransfer
 import org.eclipse.swt.widgets.Composite
 import org.eclipse.swt.widgets.Display
-import org.eclipse.ui.IPartListener
-import org.eclipse.ui.IWorkbenchPart
 import org.eclipse.ui.PlatformUI
-import org.eclipse.ui.texteditor.ITextEditor
+import org.eclipse.ui.commands.ICommandService
 import org.slf4j.LoggerFactory
+import org.testeditor.aml.AmlModel
 
 import static org.testeditor.rcp4.views.teststepselector.XtendSWTLib.*
+import org.testeditor.aml.dsl.naming.AmlQualifiedNameProvider
+import org.eclipse.core.runtime.jobs.Job
+import org.eclipse.core.runtime.jobs.JobChangeAdapter
+import org.eclipse.e4.core.services.events.IEventBroker
+import org.eclipse.core.runtime.jobs.IJobChangeEvent
 
 /** 
  * part that display a tree view with drag and drop elements of the aml model which can be inserted into
  * a tcl document
  */
 class TestStepSelector {
 
-	public static val String SELECTOR_TOPIC_REFRESH = "MaskStepSelector_Refresh"
+	public static val String SELECTOR_UPDATE_MODEL_AND_VIEW = "MaskStepSelector_Update_Model"
+	public static val String SELECTOR_UPDATE_VIEW = "MaskStepSelector_Update_View"
 
 	static val logger = LoggerFactory.getLogger(TestStepSelector);
 
+	@Inject IEventBroker broker
 	@Inject AmlInjectorProvider amlInjectorProvider
 	@Inject TestStepSelectorLabelProvider labelProvider
-	@Inject TestStepSelectorDropTextProvider dropTextProvider
 	@Inject AmlDropSupport amlDropSupport
-
-	/** set as soon as the view is populated */
-	boolean populated = false
+	@Inject TestStepSelectorDropTextProvider dropTextProvider
+	@Inject ICommandService commandService
+	TestStepSelectorTreeContentProvider contentProvider;
+	AmlQualifiedNameProvider amlQualifiedNameProvider
 	TreeViewer viewer
-
+	
+	Iterable<AmlModel> model
+	
 	@PostConstruct
-	def void postConstruct(Composite parent) {
+	def void postConstruct(Composite parent, TestStepSelectorExecutionListener executionListener,TestStepSelectorPartListener partListener) {
+		
+		val amlInjector = amlInjectorProvider.get
+		amlQualifiedNameProvider = amlInjector.getInstance(AmlQualifiedNameProvider)
+		contentProvider = amlInjectorProvider.get.getInstance(TestStepSelectorTreeContentProvider)
```

Comment:
We should set the content and label provider directly and remove them as fields.